### DISCUSSION TITLE
Button color

### DISCUSSION THREAD

**Discussion by GraceFinlay** [Telefonica/mistica-design#1022] [Proposals]
Buenas tardes,

Mi nombre es Grace y estoy trabajando en el proyecto de ecommerce global y os voy a indicar una serie de dudas relacionadas con Mística que han surgido en el desarrollo del proyecto.

# Contexto
El proyecto que está en marcha quiere incorporar unas mejoras en la tienda para que diferentes usuarios puedan comprar diferentes dispositivos. Estas mejoras que se quieren introducir son entre otras la mejora de los flujos de usuario y diferentes funcionalidades.

# Problema
Para este proyecto se requiere utilizar Mística y observamos que tenemos unas necesidades concretas en cuanto a componentes.
Es importante obtener feedback de vuestra parte con el fin de entender cómo podríamos implementar el proyecto usando Mística y qué cubra las necesidades planteadas.
En este caso queremos diseñar un botón donde el usuario seleccione el color del dispositivo que quiere en la página de detalle de producto.

# Propuesta
![Button color](https://user-images.githubusercontent.com/99174134/214359376-90bddbe3-9680-4501-a7a2-a270dc964035.png)

Para el diseño de esta button observamos lo mismo de los colores que hablamos antes pero en este caso estarían dentro de un botón secundario.

La opción de circle no logró encontrarla dentro de los assets. He visto tu ejemplo sin border radius pero no lo encuentro en el sistema. En esta ocasión actuaría el color como un icono acompañando al texto.  

Quedamos a la espera del feedback que nos puedas dar.

De antemano muchas gracias.

**Comment by aweell** [comment]
Buenas @GraceFinlay,

Si estoy entendiendo bien el contexto, este componente quizás no este conceptualmente tan cerca de un botón por comportamiento y se parezca mas a un radio group? (Entendiendo que la selección del usuario es mutuamente excluyente).

El componente chip actualmente puede configurarse como tal: 

https://mistica-web.vercel.app/?path=/story/components-chip--default

Pero no admite incluir un asset distinto al icono.

Aquí siguiendo la línea de la solución para los círculos de color de la [discussion 1015](https://github.com/Telefonica/mistica-design/discussions/1015), podría crearse un componente que funcionara de esa manera e incluyera tanto el círculo como la label.

[Ejemplo →](https://rebrand.ly/xuemhdq) 

Creemos que lo idóneo, si este es el caso, es que el componente chip pueda permitir incluir otro elemento diferente a un icono. 

Convierto esta discussion en una issue para incluir este comportamiento en la definición del componente.

**Reply by GraceFinlay** [reply]
Muchas gracias por la respuesta.
